language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/Cek.hs

```diff
@@ -316,12 +306,9 @@ runCek
     -> (Either CekEvaluationException (Plain Term), ExBudgetState)
 runCek means mode budget term =
     runCekM (CekEnv means mempty mode)
-            (ExBudgetState mempty (budget <> ExBudget 0 (modeModifier $ termAnn memTerm)))
+            (ExBudgetState mempty (budget <> ExBudget 0 (termAnn memTerm)))
```

Comment:
Shouldn't this also be a `spendBudget` call? So we can fail here if we go over, and we record it under some key.

Reply:
Yeah, just changed it locally after going over the code again.